packages/rocketchat-livechat/imports/server/rest/messages.js

```diff
@@ -0,0 +1,71 @@
+import LivechatVisitors from '../../../server/models/LivechatVisitors';
+
+function checkAuthentication(request) {
+	const receivedToken = request.headers['x-rocketchat-livechat-token'];
+	const secretToken = RocketChat.settings.get('Livechat_secret_token');
+	return receivedToken && receivedToken === secretToken;
+}
+
+RocketChat.API.v1.addRoute('livechat/status/:visitorToken', {
+	get() {
+		if (!checkAuthentication(this.request)) {
+			return RocketChat.API.v1.unauthorized();
+		}
+
+		const info = Meteor.call('livechat:getInitialData', this.urlParams.visitorToken);
```

Comment:
Which data you want to get here? This method is returning a lot of data not related to visitor's status

Reply:
@rodrigok we change this to pick only the fields below from initialData result:
* visitor
* room
* departments
* online
* agentData